timelock-agent/src/main/java/com/palantir/timelock/paxos/LegacyTimeLockServicesCreator.java

```diff
@@ -0,0 +1,88 @@
+ * http://opensource.org/licenses/BSD-3-Clause
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.timelock.paxos;
+
+import java.util.function.Supplier;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.codahale.metrics.MetricRegistry;
+import com.palantir.atlasdb.timelock.TimeLockServices;
+import com.palantir.atlasdb.timelock.paxos.ManagedTimestampService;
+import com.palantir.atlasdb.timelock.util.AsyncOrLegacyTimelockService;
+import com.palantir.atlasdb.util.AtlasDbMetrics;
+import com.palantir.lock.LockClient;
+import com.palantir.lock.RemoteLockService;
+import com.palantir.lock.impl.LegacyTimelockService;
+import com.palantir.lock.v2.TimelockService;
+
+public class LegacyTimeLockServicesCreator implements TimeLockServicesCreator {
+    private static final Logger log = LoggerFactory.getLogger(LegacyTimeLockServicesCreator.class);
+    private static final LockClient LEGACY_LOCK_CLIENT = LockClient.of("legacy");
+
+    private final PaxosLeadershipCreator leadershipCreator;
+
+    public LegacyTimeLockServicesCreator(PaxosLeadershipCreator leadershipCreator) {
+        this.leadershipCreator = leadershipCreator;
+    }
+
+    @Override
+    public TimeLockServices createTimeLockServices(
+            String client,
+            Supplier<ManagedTimestampService> rawTimestampServiceSupplier,
+            Supplier<RemoteLockService> rawLockServiceSupplier) {
+        log.info("Creating legacy timelock service for client {}", client);
+        ManagedTimestampService timestampService = instrumentInLeadershipProxy(
+                ManagedTimestampService.class,
+                rawTimestampServiceSupplier,
+                client);
+        RemoteLockService remoteLockService = instrumentInLeadershipProxy(
+                RemoteLockService.class,
+                rawLockServiceSupplier,
+                client);
+
+        // The underlying primitives are already wrapped in a leadership proxy (and must be
+        // Wrapping this means that we will make 2 paxos checks per request, which is silly.
+        TimelockService legacyTimelockService = instrument(
+                TimelockService.class,
+                createRawLegacyTimelockService(() -> timestampService, () -> remoteLockService),
+                client);
+        return TimeLockServices.create(
+                timestampService,
+                remoteLockService,
+                AsyncOrLegacyTimelockService.createFromLegacyTimelock(legacyTimelockService),
+                timestampService);
+    }
+
+    private static LegacyTimelockService createRawLegacyTimelockService(
+            Supplier<ManagedTimestampService> rawTimestampServiceSupplier,
```

Comment:
Hmm, you're right that they don't. This is only called once.
I've changed them to just use the classes.